hare/notifytracker.go

```diff
@@ -48,21 +50,17 @@ func (nt *notifyTracker) NotificationsCount(s *Set) int {
 }
 
 // calculates a unique id for the provided k and set.
-func calcID(k uint32, set *Set) uint32 {
-	hash := fnv.New32()
+func calcID(k uint32, set *Set) types.Hash32 {
+	h := hash.New()
 
 	// write K
 	buff := make([]byte, 4)
 	binary.LittleEndian.PutUint32(buff, k) // K>=0 because this not pre-round
-	hash.Write(buff)
-
-	// TODO: is this hash enough for this usage?
```

Comment:
Switching from 32bits to 256bits (32 bytes) because the set ids aren't sent over the wire and aren't stored on disk. Easier than trying to prove the absence of a vulnerability caused by a hash collision.

If Hare setIDs become a performance or memory issue, we can change algo later on without worry as they're not sent over the wire.